src/database/OfflineDataProcessor.ts

```diff
@@ -0,0 +1,183 @@
+import {DateTimeFormatter, LocalDate} from "js-joda";
+import {DatabaseConfiguration} from "./DatabaseConnection";
+import {execSync} from "child_process";
+import {faresView} from "../../config/fares/views";
+import memoize = require("memoized-class-decorator");
+import {routeingViews} from "../../config/routeing/views";
+import {timetableViews} from "../../config/timetable/views";
+import {ojpViews} from "../../config/gtfs/views";
+
+
+export class OfflineDataProcessor {
+
+  public static DATE_FORMAT = 'dd_MM';
+  private static DUMP_ONLY_STRUCTURES = ['ojp'];
+
+  private readonly execSyncOptions = {cwd: "./"};
+
+  private credentials;
+
+  /**
+   * @param databaseName
+   * @param databaseConfiguration
+   * @param commandExecutor - mainly to overwrite it in unit tests
+   */
+  public constructor(
+    private readonly databaseName: string,
+    private readonly databaseConfiguration: DatabaseConfiguration,
```

Comment:
is there any reason why we can't use database object instead of executing command line script ?

Reply:
Yup. Library which we use to connect with mysql and works strictly around specific given database and it doesn't allow us to execute CREATE DATABASE etc. believe me I've tried not to use this approach.

Reply:
what a crap ......

Reply:
Or actually it may work. Let me check how I can change it.